Compiler error on OpenSuSE Tumbleweed

<!-- Please see CONTRIBUTING.md for guidelines. -->

## Environment

* SuperCollider version: 3.11
* Operating system: OpenSuSE Tumbleweed
* Other details (Qt version, audio driver, etc.): readline 8

## Steps to reproduce

```cmake
➜  build git:(3.11) cmake -DCMAKE_INSTALL_PREFIX=/usr/local \
          -DCMAKE_PREFIX_PATH=/usr/lib/x86_64-linux-gnu/ \
          -DBUILD_TESTING=OFF -DSC_ED=OFF -DSC_EL=OFF \
          -DSC_VIM=ON -DSC_IDE=ON ..
-- SuperCollider Version: 3.11.0
-- Building from branch 3.11, commit hash is 304ff8182
-- Build type defaulting to "RelWithDebInfo"
-- Compiling with Qt GUI
-- Using bundled boost
-- Using bundled yaml-cpp
-- HIDAPI components:
--     linux hidraw
-- libudev stable: 1
-- Found UDev: /usr/lib64/libudev.so
--    include: /usr/include
--     hidapi_parser
-- Building with HID support
-- FFT library (scsynth): Green
-- Found jack: /usr/lib64/libjack.so
-- Audio API: jack
-- FFT library (supernova): Green
-- Found ALSA: /usr/lib64/libasound.so (found version "1.2.2") 
-- Found GNU Readline version 8.0: /usr/lib64/libreadline.so
-- Found Qt: /home/bja/Programme/anaconda3/lib/libQt5Core.so.5.9.7
-- sclang: Building with QtWebEngine
-- Compiling with Ableton Link support
-- Compiling with Readline support
-- Compiling with ALSA midi support
-- Building the Qt IDE
-- IDE: Building with QtWebEngine
-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/projects/build-supercollider/tmpsc/supercollider/build
```

```make
➜  build git:(3.11) make -j4
...
[ 97%] Linking CXX executable sclang
[ 97%] Linking CXX executable scide
[ 99%] Built target SuperCollider
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetnum@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `PC@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tputs@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `UP@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgoto@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetent@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetstr@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetflag@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `BC@NCURSES6_TINFO_5.0.19991023'
collect2: error: ld returned 1 exit status
make[2]: *** [lang/CMakeFiles/sclang.dir/build.make:137: lang/sclang] Fehler 1
make[1]: *** [CMakeFiles/Makefile2:2549: lang/CMakeFiles/sclang.dir/all] Fehler 2
make: *** [Makefile:183: all] Fehler 2
```

## Expected vs. actual behavior

On my system are readline 6 and readline 8 available. Has anybody an idea?

VERBOSE=1
```
[100%] Linking CXX executable sclang
cd /home/user/projects/build-supercollider/tmpsc/supercollider/build/lang && /usr/bin/cmake -E cmake_link_script CMakeFiles/sclang.dir/link.txt --verbose=1
/usr/bin/c++   -msse -mfpmath=sse -msse2 -O2 -g -DNDEBUG   CMakeFiles/sclang.dir/LangSource/cmdLineFuncs.cpp.o  -o sclang  -Wl,-rpath,/home/user/Programme/anaconda3/lib::::::::::::::: libsclang.a /usr/lib64/libpthread.so /home/user/Programme/anaconda3/lib/libQt5Sql.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5OpenGL.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Svg.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5WebEngineWidgets.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5PrintSupport.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Widgets.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5WebEngineCore.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Quick.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Gui.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5WebChannel.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Qml.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Network.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Positioning.so.5.9.7 /home/user/Programme/anaconda3/lib/libQt5Core.so.5.9.7 /usr/lib64/libX11.so ../external_libraries/hidapi/linux/libhidapi.a /usr/lib64/libudev.so ../external_libraries/hidapi/hidapi_parser/libhidapi_parser.a ../external_libraries/libboost_thread_lib.a ../external_libraries/libboost_regex_lib.a ../server/scsynth/libscsynth.a ../external_libraries/libtlsf.a ../external_libraries/libboost_system_lib.a ../external_libraries/libboost_filesystem_lib.a /usr/lib64/libdl.so /usr/lib64/libavahi-common.so /usr/lib64/libavahi-client.so /usr/lib64/libjack.so /usr/lib64/libreadline.so /usr/lib64/libasound.so /usr/lib64/libsndfile.so -lrt ../external_libraries/libyaml.a 
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetnum@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `PC@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tputs@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `UP@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgoto@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetent@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetstr@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `tgetflag@NCURSES6_TINFO_5.0.19991023'
/usr/lib64/gcc/x86_64-suse-linux/9/../../../../x86_64-suse-linux/bin/ld: /usr/lib64/libreadline.so: undefined reference to `BC@NCURSES6_TINFO_5.0.19991023'
collect2: error: ld returned 1 exit status
make[2]: *** [lang/CMakeFiles/sclang.dir/build.make:138: lang/sclang] Fehler 1
make[2]: Verzeichnis „/home/user/projects/build-supercollider/tmpsc/supercollider/build“ wird verlassen
make[1]: *** [CMakeFiles/Makefile2:2549: lang/CMakeFiles/sclang.dir/all] Fehler 2
make[1]: Verzeichnis „/home/user/projects/build-supercollider/tmpsc/supercollider/build“ wird verlassen
make: *** [Makefile:183: all] Fehler 2
```

ncurses:
```
Name               : libncurses5
Version            : 6.2.20200404-2.1
Arch               : x86_64
Name               : libncurses6
Version            : 6.2.20200404-2.1
Arch               : x86_64
```

libtinfo:
```
Name               : termcap
Version            : 2.0.8-1047.2
Arch               : x86_64
➜  build git:(3.11) l /lib64/libtin*          
lrwxrwxrwx 1 root root   15 19. Apr 22:04 /lib64/libtinfo.so.5 -> libtinfo.so.5.9
-rwxr-xr-x 1 root root 229K 19. Apr 22:04 /lib64/libtinfo.so.5.9
lrwxrwxrwx 1 root root   15 19. Apr 22:02 /lib64/libtinfo.so.6 -> libtinfo.so.6.2
-rwxr-xr-x 1 root root 194K 19. Apr 22:04 /lib64/libtinfo.so.6.2
lrwxrwxrwx 1 root root   16 26. Apr 18:18 /lib64/libtinfow.so.5 -> libtinfow.so.5.9
-rwxr-xr-x 1 root root  14K 19. Apr 22:04 /lib64/libtinfow.so.5.9
lrwxrwxrwx 1 root root   16 27. Feb 15:42 /lib64/libtinfow.so.6 -> libtinfow.so.6.2
-rwxr-xr-x 1 root root  14K 19. Apr 22:04 /lib64/libtinfow.so.6.2
```

--- a/lang/CMakeLists.txt
+++ b/lang/CMakeLists.txt
@@ -260,10 +260,18 @@ endif()
 
 ## external libraries
 if(READLINE_FOUND)
-	message(STATUS "Compiling with Readline support")
-	target_compile_definitions(libsclang PUBLIC HAVE_READLINE)
-	target_include_directories(libsclang PUBLIC ${READLINE_INCLUDE_DIR})
-	target_link_libraries(libsclang ${READLINE_LIBRARY})
+    message(STATUS "Compiling with Readline support")
+    # Some Linuxes require linking against ncurses as well as libreadline, because libtermcap provides
+    # the same API and so could be used instead. Since ncurses is widely available and usually installed
+    # already, we simply require it on all Linuxes. See #4899.
+    if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+        set(CURSES_NEED_NCURSES 1)
+        find_package(Curses REQUIRED)
+        target_link_libraries(libsclang ${CURSES_LIBRARIES})
+    endif()
+    target_compile_definitions(libsclang PUBLIC HAVE_READLINE)
+    target_include_directories(libsclang PUBLIC ${READLINE_INCLUDE_DIR})
+    target_link_libraries(libsclang ${READLINE_LIBRARY})
 endif(READLINE_FOUND)
 mark_as_advanced(READLINE_INCLUDE_DIR READLINE_LIBRARY)
 

